Mathlib/AlgebraicGeometry/ProjectiveSpectrum/Scheme.lean

```diff
@@ -578,4 +606,268 @@ def projIsoSpecTopComponent {f : A} {m : ℕ} (f_deg : f ∈ 𝒜 m) (hm : 0 < m
+
+open HomogeneousLocalization in
+lemma toStalk_specStalkEquiv (f) (x : pbo f) {m} (f_deg : f ∈ 𝒜 m) (hm : 0 < m) :
+    StructureSheaf.toStalk (A⁰_ f) ((toSpec 𝒜 f).1.base x) ≫ (specStalkEquiv 𝒜 f x f_deg hm).hom =
+      (mapId _ <| Submonoid.powers_le.mpr x.2 : (A⁰_ f) →+* AtPrime 𝒜 x.1.1.toIdeal) :=
+  letI : Algebra (Away 𝒜 f) (AtPrime 𝒜 x.1.asHomogeneousIdeal.toIdeal) :=
+    (mapId 𝒜 (Submonoid.powers_le.mpr x.2)).toAlgebra
+  letI := isLocalization_atPrime 𝒜 f x f_deg hm
+  (IsLocalization.algEquiv
+    (R := A⁰_ f)
+    (M := ((toSpec 𝒜 f).1.base x).asIdeal.primeCompl)
+    (S := (Spec.structureSheaf (A⁰_ f)).presheaf.stalk ((toSpec 𝒜 f).1.base x))
+    (Q := AtPrime 𝒜 x.1.asHomogeneousIdeal.toIdeal)).toAlgHom.comp_algebraMap
+
+open HomogeneousLocalization in
+lemma stalkMap_toSpec (f) (x : pbo f) {m} (f_deg : f ∈ 𝒜 m) (hm : 0 < m) :
+    PresheafedSpace.stalkMap (toSpec 𝒜 f).1 x =
+      (specStalkEquiv 𝒜 f x f_deg hm).hom ≫ (Proj.stalkIso' 𝒜 x.1).toCommRingCatIso.inv ≫
+      ((Proj.toLocallyRingedSpace 𝒜).restrictStalkIso (Opens.openEmbedding _) x).inv := by
+  apply IsLocalization.ringHom_ext (R := A⁰_ f) ((toSpec 𝒜 f).1.base x).asIdeal.primeCompl
+    (S := (Spec.structureSheaf (A⁰_ f)).presheaf.stalk ((toSpec 𝒜 f).1.base x))
+  refine (toStalk_stalkMap_toSpec _ _ _).trans ?_
+  rw [awayToΓ_ΓToStalk, ← toStalk_specStalkEquiv 𝒜 f x f_deg hm, Category.assoc]
+  rfl
+
+lemma isIso_toSpec (f) {m} (f_deg : f ∈ 𝒜 m) (hm : 0 < m) :
+    IsIso (toSpec 𝒜 f) := by
+  have (x) : IsIso (PresheafedSpace.stalkMap (toSpec 𝒜 f).1 x) := by
+    rw [stalkMap_toSpec 𝒜 f x f_deg hm]; infer_instance
+  have : LocallyRingedSpace.IsOpenImmersion (toSpec 𝒜 f) := by
+    apply SheafedSpace.IsOpenImmersion.of_stalk_iso
+    convert (TopCat.homeoOfIso (projIsoSpecTopComponent f_deg hm)).openEmbedding using 1
+    ext; exact ProjIsoSpec.toSpec_eq_toSpec 𝒜 _
+  suffices IsIso (LocallyRingedSpace.forgetToSheafedSpace.map (toSpec 𝒜 f)) by
+    apply isIso_of_reflects_iso _ LocallyRingedSpace.forgetToSheafedSpace
+  show IsIso (toSpec 𝒜 f).1
+  suffices IsIso (SheafedSpace.forgetToPresheafedSpace.map (toSpec 𝒜 f).1) by
+    apply isIso_of_reflects_iso _ SheafedSpace.forgetToPresheafedSpace
+  suffices Epi (SheafedSpace.forgetToPresheafedSpace.map (toSpec 𝒜 f).val).base by
+    exact PresheafedSpace.IsOpenImmersion.to_iso _
+  rw [TopCat.epi_iff_surjective]
+  convert (TopCat.homeoOfIso (projIsoSpecTopComponent f_deg hm)).surjective using 1
+  ext; exact ProjIsoSpec.toSpec_eq_toSpec 𝒜 _
+
+end ProjIsoSpec
+
+/--
+If `f ∈ A` is a homogeneous element of positive degree, then the projective spectrum restricted to
+`D(f)` as a locally ringed space is isomorphic to `Spec A⁰_f`.
+-/
+def projIsoSpec (f) {m} (f_deg : f ∈ 𝒜 m) (hm : 0 < m) :
+    (Proj| pbo f) ≅ (Spec (A⁰_ f)) :=
+  @asIso (f := ProjIsoSpec.toSpec 𝒜 f) (ProjIsoSpec.isIso_toSpec 𝒜 f f_deg hm)
+
+/--
+For any `ℕ`-graded ring `A`, the projective spectrum of `A` is locally affine.
```

Comment:
```suggestion
This is the scheme `Proj(A)` for any `ℕ`-graded ring `A`.
```
Could you add a notation for this scheme?

Reply:
I think this can be called `Proj` directly among all the other (algebro-geometric) Proj constructions?

Reply:
I am not very good with notations, but, I feel it would be reasonable to expect that the `Proj` notation for this applies at least if the user has opened `AlgebraicGeometry`.